processing/src/main/java/org/apache/druid/segment/ReferenceCountingSegment.java

```diff
@@ -167,6 +166,30 @@ public void close()
     }
   }
 
+  public ReferenceCounter referenceCounter()
+  {
+    return new ReferenceCounter()
+    {
+      @Override
+      public boolean increment()
+      {
+        return ReferenceCountingSegment.this.increment();
+      }
+
+      @Override
+      public Closeable decrementOnceCloseable()
+      {
+        return ReferenceCountingSegment.this.decrementOnceCloseable();
+      }
+
+      @Override
+      public void decrement()
+      {
+        ReferenceCountingSegment.this.decrement();
+      }
+    };
```

Comment:
Make `ReferenceCountingSegment` implement `ReferenceCounter`. Then this function can just be `return this;`

Which also means less inner classes and less lines of code to read in a delta 🙃

Reply:
Oh, great idea!